src/ocaml/typing/408/typemod.ml

```diff
@@ -2529,21 +2527,10 @@ let type_package env m p nl =
   in
   let tl' =
     List.map
-      (fun name ->
-         let path = mkpath mp name in
-         match Env.find_type path env with
-         | decl -> begin
-             match decl.type_params with
-             | [] -> Btype.newgenty (Tconstr (path, [], ref Mnil))
-             | _ ->
-               raise (Error(m.pmod_loc, env,
-                            Package_type_arity(p, modl.mod_type, name)))
-           end
-         | exception Not_found ->
-             raise (Error(m.pmod_loc, env,
-                          Package_type_missing(p, modl.mod_type, name))))
-      nl
-  in
+      (fun name -> Btype.newgenty (Tconstr (mkpath mp name,[],ref Mnil)))
+      (* beware of interactions with Printtyp and short-path:
+         mp.name may have an arity > 0, cf. PR#7534 *)
+      nl in
```

Comment:
Here you're reverting f670d19f8097ab06ffd1d27be6158c14c77f9e12 .
The initial bug seems to have been fixed upstream in 4.06 ( https://github.com/ocaml/ocaml/pull/1186 ). Can you confirm that it is indeed fixed in merlin, and then do the change in the 4.06 and 4.07 backends as well?

Reply:
No, it seems the bug persists. I will revert this change.